Invalid sprintf format in objects_timeperiod

timerange2str uses int for the calculations, but %02hd for short in the output format. The clang compiler doesn't allow this and exits with an error:

```
src/naemon/objects_timeperiod.c:289:42: error: format specifies type 'short' but the argument has type 'int' [-Werror,-Wformat]
        sprintf(str, "%02hd:%02hd-%02hd:%02hd", sh, sm, eh, em);
                      ~~~~~                     ^~
                      %02d
src/naemon/objects_timeperiod.c:289:46: error: format specifies type 'short' but the argument has type 'int' [-Werror,-Wformat]
        sprintf(str, "%02hd:%02hd-%02hd:%02hd", sh, sm, eh, em);
                            ~~~~~                   ^~
                            %02d
src/naemon/objects_timeperiod.c:289:50: error: format specifies type 'short' but the argument has type 'int' [-Werror,-Wformat]
        sprintf(str, "%02hd:%02hd-%02hd:%02hd", sh, sm, eh, em);
                                  ~~~~~                 ^~
                                  %02d
src/naemon/objects_timeperiod.c:289:54: error: format specifies type 'short' but the argument has type 'int' [-Werror,-Wformat]
        sprintf(str, "%02hd:%02hd-%02hd:%02hd", sh, sm, eh, em);
                                        ~~~~~               ^~
                                        %02d
4 errors generated.
Makefile:1920: recipe for target 'src/naemon/objects_timeperiod.lo' failed
```

It compiles if I change the sprintf to this:
`sprintf(str, "%02d:%02d-%02d:%02d", sh, sm, eh, em);`

--- a/src/naemon/objects_timeperiod.c
+++ b/src/naemon/objects_timeperiod.c
@@ -286,7 +286,7 @@ static const char *timerange2str(const timerange *tr)
 	sm = (tr->range_start / 60) % 60;
 	eh = tr->range_end / 3600;
 	em = (tr->range_end / 60) % 60;
-	sprintf(str, "%02hd:%02hd-%02hd:%02hd", sh, sm, eh, em);
+	sprintf(str, "%02d:%02d-%02d:%02d", sh, sm, eh, em);
 	return str;
 }
 

